src/Service.Tests/Unittests/ConfigFileWatcherUnitTests.cs

```diff
@@ -0,0 +1,143 @@
+      ""allow-introspection"": " + initialGQLIntrospection.ToString().ToLower() + @"
+    },
+    ""host"": {
+      ""cors"": {
+        ""origins"": [
+          ""http://localhost:5000""
+        ],
+        ""allow-credentials"": false
+      },
+      ""authentication"": {
+        ""provider"": ""StaticWebApps""
+      },
+      ""mode"": """ + initialMode + @"""
+    }
+  },
+  ""entities"": {}
+}";
+            string updatedConfig = @"
+{
+  ""$schema"": ""https://github.com/Azure/data-api-builder/releases/download/vmajor.minor.patch/dab.draft.schema.json"",
+  ""data-source"": {
+    ""database-type"": ""mssql"",
+    ""connection-string"": ""Server=test;Database=test;User ID=test;Password=test;"",
+    ""options"": {
+      ""set-session-context"": true
+    }
+  },
+  ""runtime"": {
+    ""rest"": {
+      ""enabled"": " + updatedRestEnabled.ToString().ToLower() + @",
+      ""path"": """ + updatedRestPath + @"""
+    },
+    ""graphql"": {
+      ""enabled"": " + updatedGQLEnabled.ToString().ToLower() + @",
+      ""path"": """ + updatedGQLPath + @""",
+      ""allow-introspection"": " + updatedGQLIntrospection.ToString().ToLower() + @"
+    },
+    ""host"": {
+      ""cors"": {
+        ""origins"": [
+          ""http://localhost:5000""
+        ],
+        ""allow-credentials"": false
+      },
+      ""authentication"": {
+        ""provider"": ""StaticWebApps""
+      },
+      ""mode"": """ + updatedMode + @"""
+    }
+  },
+  ""entities"": {}
+}";
+            string configName = "config.json";
+            if (File.Exists(configName))
+            {
+                File.Delete(configName);
+            }
+
+            FileSystem fileSystem = new();
+            fileSystem.File.WriteAllText(configName, initialConfig);
+            FileSystemRuntimeConfigLoader configLoader = new(fileSystem, configName, string.Empty);
+            RuntimeConfigProvider configProvider = new(configLoader);
+            // Must GetConfig() to start file watching
+            RuntimeConfig runtimeConfig = configProvider.GetConfig();
+            Assert.IsNotNull(runtimeConfig);
+            // Simulate change to the config file
+            fileSystem.File.WriteAllText(configName, updatedConfig);
+            // Give file watcher enough time to see the change
+            System.Threading.Thread.Sleep(1000);
+            // Hot Reloaded config
+            runtimeConfig = configProvider.GetConfig();
+            Assert.AreEqual(updatedRestEnabled, runtimeConfig.Runtime.Rest.Enabled);
+            Assert.AreEqual(updatedRestPath, runtimeConfig.Runtime.Rest.Path);
+            Assert.AreEqual(updatedGQLEnabled, runtimeConfig.Runtime.GraphQL.Enabled);
+            Assert.AreEqual(updatedGQLPath, runtimeConfig.Runtime.GraphQL.Path);
+            Assert.AreEqual(updatedGQLIntrospection, runtimeConfig.Runtime.GraphQL.AllowIntrospection);
+            Assert.AreEqual(updatedMode, runtimeConfig.Runtime.Host.Mode);
```

Comment:
this asserts tells us that config file is changed.
Can we do something like hitting a rest endpoint before hot reload and then change the endpoint and then try to hit the updated endpoint to show it's updated.

Reply:
that's a good idea for integration testing, ill figure out a way to add some tests like that

Reply:
integration testing introduces some new challenges with race conditions. I have opened an issue to track adding these tests, they will come in a future PR.